front-end/src/components/LoanDetails/LoanDetails.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useContext, useEffect } from 'react'
+import { Content, Container, Spinner } from 'native-base'
+import { connect } from 'react-redux'
+import { NavigationContext } from 'react-navigation'
+import PropTypes from 'prop-types'
+
+import Detail from './Detail'
+import Chart from './Chart'
+import History from './History'
+
+import { fetchDetail } from '../../lib/actions'
+
+function LoanDetails({ fetchData, loanList, loading }) {
+  const [loan, setLoan] = useState({})
+  const navigation = useContext(NavigationContext)
+  const { id } = navigation.getParam('shortLoanData')
+
+  useEffect(() => {
+    fetchData(id)
+  }, [])
+
+  useEffect(() => {
+    setLoan(loanList.find(item => item.id === id))
+  }, [loanList])
+
+  return (
+    <Container>
+      <Content>
+        <Detail />
+        {loading ? (
+          <Spinner color="blue" />
+        ) : (
+          <>
+            <Chart loan={loan} />
```

Comment:
@Jeffzholy and for these you don't need to pass anything, just the history list is going to be enough to be mapped from the state